src/js/rtp-stats.js

```diff
@@ -3,118 +3,109 @@
+
+                } else if (typeof statsReport.packetsReceived !== 'undefined' && statsReport.mediaType == 'video' && streamType === 'video_output') {
+                    extractedStats = {
+                        timestamp:          timestamp,
+                        packetsSent:        statsReport.packetsSent,
+                        bytesSent:          statsReport.bytesSent,
+                        packetsLost:        is_defined(statsReport.packetsLost) ? Math.max(0, statsReport.packetsLost) : 0,
+                        frameRateReceived:  when_defined(statsReport.googFrameRateReceived),
+                        procMilliseconds:   is_defined(statsReport.googCurrentDelayMs),
+                        jbMilliseconds:     when_defined(statsReport.googJitterBufferMs)
+                    };
+
+                }
+            } else if (statsReport.type === 'inboundrtp') {
+                // Firefox case. Firefox reports packetsLost parameter only in inboundrtp type, and doesn't report in outboundrtp type.
+                // So we only pull from inboundrtp. Firefox reports only stats for the stream passed in.
+                if (is_defined(statsReport.packetsLost) && is_defined(statsReport.packetsReceived)) {
+                    extractedStats = {
+                        packetsLost:        statsReport.packetsLost,
+                        packetsReceived:    statsReport.packetsReceived,
+                        audioLevel:         when_defined(statsReport.audioInputLevel),
+                        rttMilliseconds:    when_defined(statsReport.mozRtt),
+                        jbMilliseconds:     when_defined(statsReport.jitter)
+                    };
                 }
             }
         }
     }
-    return callStats;
+
+    return extractedStats ? new MediaRtpStats(extractedStats) : null;
 }
 
 /**
 * Basic RTP statistics object, represents statistics of an audio or video stream.
 */
 class MediaRtpStats {
-    constructor(timestamp, packetsLost, packetsCount, audioLevel, rttMilliseconds, jbMilliseconds, bytesSent, bytesReceived, framesEncoded, framesDecoded, frameRateSent, frameRateReceived) {
-        this._timestamp = timestamp;
-        this._packetsLost = packetsLost;
-        this._packetsCount = packetsCount;
-        this._audioLevel = audioLevel;
-        this._rttMilliseconds = rttMilliseconds;
-        this._jbMilliseconds = jbMilliseconds;
-        this._bytesSent = bytesSent;
-        this._bytesReceived = bytesReceived;
-        this._framesEncoded = framesEncoded;
-        this._framesDecoded = framesDecoded;
-        this._frameRateSent = frameRateSent;
-        this._frameRateReceived = frameRateReceived;
+    constructor(paramsIn) {
+        var params = paramsIn || {};
+
+        this._timestamp = params.timestamp || new Date().getTime();
+        this._packetsLost = params.packetsLost || 0;
+        this._packetsCount = params.packetsCount || 0;
+        this._audioLevel = params.audioLevel || 0;
```

Comment:
for some reason fi packetsCount and audioLevel are not extracted from browser stats we should keep them null and not make it 0 I think. It might convey wrong information.